kotlinx-coroutines-core/nativeDarwin/src/Dispatchers.kt

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2016-2021 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines
+
+import kotlinx.cinterop.*
+import kotlinx.coroutines.internal.*
+import platform.CoreFoundation.*
+import platform.darwin.*
+import kotlin.coroutines.*
+import kotlin.native.concurrent.*
+import kotlin.native.internal.NativePtr
+
+internal fun isMainThread(): Boolean = CFRunLoopGetCurrent() == CFRunLoopGetMain()
+
+internal actual fun createMainDispatcher(default: CoroutineDispatcher): MainCoroutineDispatcher =
+    if (multithreadingSupported) DarwinMainDispatcher(false) else OldMainDispatcher(Dispatchers.Default)
+
+internal actual fun createDefaultDispatcher(): CoroutineDispatcher = DarwinGlobalQueueDispatcher
+
+private object DarwinGlobalQueueDispatcher : CoroutineDispatcher() {
+    override fun dispatch(context: CoroutineContext, block: Runnable) {
+        platformAutoreleasePool {
+            dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT.convert(), 0)) {
+                block.run()
+            }
+        }
+    }
+}
+
+private class DarwinMainDispatcher(
+    private val invokeImmediately: Boolean
+) : MainCoroutineDispatcher(), Delay {
+    
+    override val immediate: MainCoroutineDispatcher =
+        if (invokeImmediately) this else DarwinMainDispatcher(true)
+
+    override fun isDispatchNeeded(context: CoroutineContext): Boolean = !(invokeImmediately && isMainThread())
+
+    override fun dispatch(context: CoroutineContext, block: Runnable) {
+        platformAutoreleasePool {
```

Comment:
Could you please elaborate on the point of this change? 
`platformAutoreleasePool` is at least supported in IDE

Reply:
It's purely a stylistic preference, not something functional. I was somehow convinced that this is obviously more pleasing and didn't require an explanation, but since it does, feel free to ignore it.

* If we look not just at this line, but around it, we see non-Kotlin-idiomatic code, with names like `dispatch_async`. Using a name like `platformAutoreleasePool` in the block of code where everything else looks like Swift is somewhat staggering, as is any place where constructs from different ecosystems are freely mixed. `Timer` below is nice: it's internally consistent in being "purely Foundations in Kotlin syntax".
* More succinct.

So, less letters, more local stylistic consistency.